crates/sui-graphql-rpc/src/types/transaction_block_kind/consensus_commit_prologue.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    context_data::db_data_provider::PgManager,
+    types::{date_time::DateTime, epoch::Epoch},
+};
+use async_graphql::*;
+use fastcrypto::encoding::{Base58, Encoding};
+use sui_types::{
+    digests::ConsensusCommitDigest,
+    messages_checkpoint::CheckpointTimestamp,
+    messages_consensus::{
+        ConsensusCommitPrologue as NativeConsensusCommitPrologueTransactionV1,
+        ConsensusCommitPrologueV2 as NativeConsensusCommitPrologueTransactionV2,
+    },
+};
+
+/// Unlike other transaction kinds, which wrap a native transaction kind, this transaction lists out
+/// all its fields. This is because there are multiple versions of this transaction type in the
+/// protocol, and this type merges both fields, with fields that only appear in one version being
```

Comment:
What does it mean `merges both fields`? Which fields does this refer to, because there are four fields in the type definition. I assume it means merges both fields of the whatever version of this transaction, with fields that appear in one version being optional. 
So which version is optional? Kinda confused about this.

Reply:
Yeah, this comment is not clear 😅.  I'll reword it, but what it's saying is that while other transaction kinds are usually represented by directly wrapping the native representation of the kind, we can't do that here, because there are two versions of `ConsensusCommitPrologue` in the protocol.  They share most fields, but V2 has an extra field (the consensus commit digest), so we represent it by a single struct, which contains all the fields that are common to both versions, and then the `consensus_commit_digest` as optional.